packages/@react-aria/utils/src/isElementVisible.ts

```diff
@@ -63,9 +63,9 @@ function isAttributeVisible(element: Element, childElement?: Element) {
  */
 export function isElementVisible(element: Element, childElement?: Element): boolean {
   if (supportsCheckVisibility) {
-    return element.checkVisibility();
+    return element.checkVisibility() && !element.closest('[data-react-aria-prevent-focus]');
```

Comment:
not convinced this is where the check should be given the name of the function vs the name of the attribute...

Reply:
yeah this is a bit confusing cuz of the naming, but if I understand this correctly, this is to make it so isTabbable/isFocusable don't resolve as true for elements with the above data attribute correct? Any reason we don't move this check into those functions?

Reply:
It'd be annoying to do it there because you have to check all the parents as well. I think it's ok here for now